lib/jack_tokenizer.rb

```diff
@@ -69,8 +68,8 @@ def string_val
   private
 
   def skip_whitespace
-    while WHITESPACES.include?(@input[@index])
-      @index += 1
+    if (match = @input.match(%r{[ \n]*}, @index)) && match.begin(0) == @index
```

Comment:
Trivia: the [`\s` metacharacter](https://ruby-doc.org/core-3.0.3/Regexp.html#class-Regexp-label-Character+Classes) (`s` for “space”) is equivalent to the slightly broader character class `[ \t\r\n\f\v]`, so we could’ve used `%r{\s*}` here.